src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -786,4 +786,42 @@
       it { is_expected.to redirect_to(action: :show, project: user.home_project) }
     end
   end
+
+  describe "POST #add_maintained_project" do
+    before do
+      login user
+    end
+
+    context "maintained project successfully added" do
+      before do
+        user.home_project.update(kind: 'maintenance')
+      end
+
+      context "maintained project successfully added" do
+        before do
+          post :add_maintained_project, project: user.home_project, maintained_project: user.home_project.name
+        end
+
+        it { expect(user.home_project.maintained_projects.where(project_id: user.home_project.id)).to exist }
```

Comment:
You can query it by the project, without the `.id`:

`it { expect(user.home_project.maintained_projects.where(project: user.home_project)).to exist }`